pallets/liquid-staking/src/lib.rs

```diff
@@ -491,7 +507,7 @@ pub mod pallet {
             }
 
             let (bond_amount, rebond_amount, unbond_amount) =
-                MatchingPool::<T>::take().matching::<Self>(unbonding_amount)?;
+                MatchingPool::<T>::get().matching::<Self>(unbonding_amount)?;
```

Comment:
can we get only once?

Reply:
where do double reads come from, as I can see, there is only one `get` in `settlement` method